helix-lsp/src/client.rs

```diff
@@ -188,11 +188,11 @@ impl Client {
     }
 
     /// Reply to a language server RPC call.
-    pub fn reply(
+    pub fn reply<R>(
```

Comment:
There's a reason the method did not use generics: due to monomorphization it can lead to code bloat since the method is big. The way to fix this would be to rename the old function as `reply_impl`, then define a new `reply` function  that uses generics and calls `reply_impl`.

See the example here under "inlining in practice" (under "fifth,") https://matklad.github.io/2021/07/09/inline-in-rust.html